documentation/functional_testing/architecture.md

```
@@ -0,0 +1,269 @@
+
+If `A` is a page we can say that:
+- `A` is the page of `B`
+
+If a `Page`/`Element` `A` class returned an `Element` `B` we can say that:
+- `A` generated `B`
+- `B` has been generated by `A`
+
+### Element Access Rules (also in ./utils)
+
+1. **A Page/Element object MUST NEVER be able to return 2 HTML elements that overlap each other**. Let's say that `A` is any `Element` object/Locator returned by a class `Class` and `OTHERS` are all the other `Element` objects/Locators that can be returned from `Class` then:
+  1. `A` Must not be present in `OTHERS` (no HTML element in `OTHERS` must be the same as `A`).
```

markdownlint-cli2 (0.17.2)

Inconsistent indentation for list items at the same level
Expected: 0; Actual: 2

(MD005, list-indent)